### PR TITLE
GoogleStackdriverTarget - Avoid InvalidCastException for custom IConvertible

### PR DESCRIPTION
**Pull request by snakefoot**
When it returns `TypeCode.Boolean` (Ex JSON.NET JValue). Follow up to previous PR #2384

### CODE REVIEW

**Comment by jskeet** [comment]
Assigned to Chris as more of an NLog person.

**Comment by snakefoot** [comment]
Thought Chris was a log4net-person, so many boxes :)

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs [hunk 1]
```diff
@@ -161,7 +161,6 @@ protected override void InitializeTarget()
                                 break;  // Let StringEnumConverter handle formatting
                             return Value.ForNumber(Convert.ToDouble(o));
                         case TypeCode.String:
-                            return Value.ForString((string)o);
                         case TypeCode.Char:
                             return Value.ForString(o.ToString());
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs]: Should this use `Convert.ToString(o))`?

Comment (snakefoot) [reply apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs]: Thought object.ToString would be good enough.

**Comment by snakefoot** [comment]
Looks like serviceContext should be included according to docs:

https://cloud.google.com/error-reporting/docs/formatting-error-messages#json_representation

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs (hunk 1 above)
Comment (chrisdunelm) [reply apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs]: OK

**Comment by chrisdunelm** [comment]
(Sorry for the delay)
Please put the second commit into a different PR; it's unrelated to the conversion change.
Once that's done, the first change can be merged.

**Comment by snakefoot** [comment]
@chrisdunelm Have reverted the last commit. And will make a new PR when this has been merged.

**Comment by chrisdunelm** [comment]
Thanks :)

**Pull request merged**